src/nvapi_vulkan.cpp

```diff
@@ -0,0 +1,255 @@
+    }
+
+    NvAPI_Status __cdecl NvAPI_Vulkan_SetSleepMode(HANDLE vkDevice, NV_VULKAN_SET_SLEEP_MODE_PARAMS* pSetSleepModeParams) {
+        constexpr auto n = __func__;
+        thread_local bool alreadyLoggedOk = false;
+
+        if (log::tracing())
+            log::trace(n, log::fmt::ptr(vkDevice), log::fmt::nv_vk_set_sleep_status_params(pSetSleepModeParams));
+
+        auto device = reinterpret_cast<VkDevice>(vkDevice);
+
+        if (!device)
+            return InvalidArgument(n);
+
+        if (pSetSleepModeParams && pSetSleepModeParams->version != NV_VULKAN_SET_SLEEP_MODE_PARAMS_VER1)
+            return IncompatibleStructVersion(n, pSetSleepModeParams->version);
+
+        auto lowLatencyDevice = NvapiVulkanLowLatencyDevice::Get(device);
+
+        if (!lowLatencyDevice)
+            return HandleInvalidated(n);
+
+        auto vr = pSetSleepModeParams
+            ? lowLatencyDevice->SetLatencySleepMode(pSetSleepModeParams->bLowLatencyMode, pSetSleepModeParams->bLowLatencyBoost, pSetSleepModeParams->minimumIntervalUs)
+            : lowLatencyDevice->SetLatencySleepMode(nullptr);
+
+        return vr == VK_SUCCESS ? Ok(n, alreadyLoggedOk) : Error(n, vr);
+    }
+
+    NvAPI_Status __cdecl NvAPI_Vulkan_Sleep(HANDLE vkDevice, NvU64 signalValue) {
+        constexpr auto n = __func__;
+        thread_local bool alreadyLoggedOk = false;
+
+        if (log::tracing())
+            log::trace(n, log::fmt::ptr(vkDevice), signalValue);
+
+        auto device = reinterpret_cast<VkDevice>(vkDevice);
+
+        if (!device)
+            return InvalidArgument(n);
+
+        auto lowLatencyDevice = NvapiVulkanLowLatencyDevice::Get(device);
+
+        if (!lowLatencyDevice)
+            return HandleInvalidated(n);
+
+        auto vr = lowLatencyDevice->LatencySleep(signalValue);
+
+        return vr == VK_SUCCESS ? Ok(n, alreadyLoggedOk) : Error(n, vr);
+    }
+
+    NvAPI_Status __cdecl NvAPI_Vulkan_GetLatency(HANDLE vkDevice, NV_VULKAN_LATENCY_RESULT_PARAMS* pGetLatencyParams) {
+        constexpr auto n = __func__;
+        thread_local bool alreadyLoggedOk = false;
+
+        if (log::tracing())
+            log::trace(n, log::fmt::ptr(vkDevice), log::fmt::nv_vk_latency_result_params(pGetLatencyParams));
+
+        auto device = reinterpret_cast<VkDevice>(vkDevice);
+
+        if (!device)
+            return InvalidArgument(n);
+
+        if (!pGetLatencyParams)
+            return InvalidPointer(n);
+
+        if (pGetLatencyParams->version != NV_VULKAN_LATENCY_RESULT_PARAMS_VER1)
+            return IncompatibleStructVersion(n, pGetLatencyParams->version);
+
+        auto lowLatencyDevice = NvapiVulkanLowLatencyDevice::Get(device);
+
+        if (!lowLatencyDevice)
+            return HandleInvalidated(n);
+
+        static constexpr auto count = sizeof(pGetLatencyParams->frameReport) / sizeof(*pGetLatencyParams->frameReport);
+        static_assert(count == 64);
+
+        std::array<VkLatencyTimingsFrameReportNV, count> timings;
```

Comment:
Depending on how often this is called, may be wise to initialize this for safety: 
`std::array<VkLatencyTimingsFrameReportNV, count> timings{}; `

Reply:
It might be called every frame, and this thing is unfortunately quite huge. IIRC when Strudel was testing old flavor of this in Path of Exile II, presence of latency reports actually had negative impact on FPS overall.

I think it's enough that `NvapiVulkanLowLatencyDevice` ensures that important members (`.sType` and `.pNext`) are initialized before calling Vulkan function that fills the rest.

Reply:
Alright, thanks for additional context and agree to keep your version.